packages/lit-dev-content/site/docs/v3/data/context.md

```diff
@@ -281,6 +281,21 @@ and the ContextConsumer controller:
   );
 ```
 
+### SSR
+
+Basic support for server side rendering can be enabled by assigning
+`globalThis.litSsrCallConnectedCallback = true`. This will configure `@lit-labs/ssr` to call
+`connectedCallback()` and `ReactiveController.hostConnected()`, which allows `@lit/context`
+to register the necessary event handlers for SSR.
+
+As an alternative to `document.documentElement` or `document.body` (which are expected to be undefined
+in the server environment) for global event listeners (e.g. for `@lit/context ContextProvider`), you can
+use the global variable `globalThis.litServerRoot` which is (only) available during SSR.
+
+```ts
+  new ContextProvider(isServer ? globalThis.litServerRoot : document.body, {...});
```

Comment:
the overloaded constructor signature for ContextProvider is

```ts
  constructor(host: HostElement, options: Options<T>);
  /** @deprecated Use new ContextProvider(host, options) */
  constructor(host: HostElement, context: T, initialValue?: ContextType<T>);
  constructor(
    host: HostElement,
    contextOrOptions: T | Options<T>,
    initialValue?: ContextType<T>
  ) { /*...* }
```
ContextRoot does not have any constructor params.

Do you mean something like this?

```ts
  const root = new ContextRoot();
  if (!isServer) {
    root.attach(document.body);
  } else {
    root.attach(globalThis.litServerRoot);
  }
```

Reply:
No, I am referring to the use case of registering a ContextProvider globally. (See https://www.npmjs.com/package/@lit/context and search for `document.body`)
ContextRoot _should_ not be necessary in an SSR environment, as classes are loaded in order (from my understanding).

But thank you for pointing out, that my example is incomplete and easily misunderstood. I will extend it. 😄 👍